packs/mssql/actions/lib/results_processor.py

```diff
@@ -0,0 +1,104 @@
+import _mssql
+from tempfile import NamedTemporaryFile
+import csv
+import os
+import sys
+
+__all__ = [
+    'ResultsProcessor'
+]
+
+
+# noinspection PyUnusedLocal
+# noinspection PyMethodMayBeStaticInspection
+class ResultsProcessor(object):
+    """
+    Utility for processing action response, e.g. changing returned results or passing data via disk
+    """
+
+    NO_DATA = 2
+
+    def __init__(self, config, logger):
+        self.config = config
+        self.logger = logger
+
+    def execute_scalar(self, response, cursor):
+        """
+        Returns the scalar response.
+        """
+        if not response:
+            sys.exit(self.NO_DATA)
```

Comment:
You could also throw an exception here - this will also cause action to fail, but it will make it more obvious to the user what is going on because exception can contain an error message.